tests/queries/test_query.py

```diff
@@ -160,6 +160,11 @@ def test_filter_non_conditional(self):
         with self.assertRaisesMessage(TypeError, msg):
             query.build_where(Func(output_field=CharField()))
 
+    def test_use_correct_db(self):
```

Comment:
This tests the implementation without covering the original issue as raised in the ticket, I think having something like:

```diff
diff --git a/tests/multiple_database/tests.py b/tests/multiple_database/tests.py
index 9587030a46..acda154f48 100644
--- a/tests/multiple_database/tests.py
+++ b/tests/multiple_database/tests.py
@@ -2,12 +2,12 @@ import datetime
 import pickle
 from io import StringIO
 from operator import attrgetter
-from unittest.mock import Mock
+from unittest.mock import Mock, patch

 from django.contrib.auth.models import User
 from django.contrib.contenttypes.models import ContentType
 from django.core import management
-from django.db import DEFAULT_DB_ALIAS, router, transaction
+from django.db import DEFAULT_DB_ALIAS, router, transaction, connections
 from django.db.models import signals
 from django.db.utils import ConnectionRouter
 from django.test import SimpleTestCase, TestCase, override_settings
@@ -1235,6 +1235,21 @@ class QueryTestCase(TestCase):
         val = Book.objects.raw("SELECT id FROM multiple_database_book").using("other")
         self.assertQuerySetEqual(val, [dive.pk], attrgetter("pk"))

+    def test_query_string(self):
+        with (
+            patch.object(connections["default"].ops, "compiler")
+            as default_db_compiler
+        ):
+            queryset = Person.objects.using("other").all().order_by("id")
+            self.assertEqual(
+                str(queryset.query),
+                'SELECT "multiple_database_person"."id", '
+                '"multiple_database_person"."name" '
+                'FROM "multiple_database_person" '
+                'ORDER BY "multiple_database_person"."id" ASC'
+            )
+            default_db_compiler.assert_not_called()
+
     def test_select_related(self):
         """
```
might be better

Reply:
You're right, I have add your test in the new commit.